src/app/modules/entities/state-entity-list/state-entity-list.component.html

```diff
@@ -0,0 +1,75 @@
+        </mat-icon>
+      </a>
+      <div class="list-menu">
+        <mat-form-field class="custom-form-field">
+          <mat-label>Search for State</mat-label>
+          <input matInput (keyup)="applyFilter($event)">
+          <!-- <input [(ngModel)]="search" [appDebounceDirective]="search" (onValueChange)="getOrganisationList()"
+            #searchInput matInput> -->
+          <button mat-icon-button matPrefix>
+            <mat-icon>search</mat-icon>
+          </button>
+        </mat-form-field>
+      </div>
+    </div>
+
+    <div>
+      <!-- <mat-button-toggle-group #toggleGroup="matButtonToggleGroup">
+        <mat-button-toggle matTooltip="All organisations" checked (click)="allOrganisation(toggleGroup.value)"
+          class="filter" value="All">All
+        </mat-button-toggle>
+        <mat-button-toggle matTooltip="Active organisations" (click)="allOrganisation(toggleGroup.value)" class="filter"
+          value="Active">Active
+        </mat-button-toggle>
+        <mat-button-toggle matTooltip="Inactive organisations" (click)="allOrganisation(toggleGroup.value)"
+          class="filter" value="Inactive">Inactive
+        </mat-button-toggle>
+      </mat-button-toggle-group> -->
+    </div>
+
+    <div class="row" style="padding-right: 3%;">
+      <!-- <div class="btn-gap">
+        <button mat-button (click)="addNewOrganisation()" matTooltip="Add Organisation">
+          Add Organisation
+        </button>
+      </div> -->
+    </div>
+  </div>
+
+  <div class="table-container mat-elevation-z8">
+    <app-spinner *ngIf="listing == false"></app-spinner>
+    <mat-table *ngIf="dataSource" [dataSource]="dataSource" matSort>
+      <span *ngFor="let column of columns">
+        <ng-container [matColumnDef]="column.key">
+          <mat-header-cell *matHeaderCellDef>{{ column.label}}</mat-header-cell>
+          <span *ngIf="column.type == 'column'">
+            <mat-cell *matCellDef="let row" class="text-capital">
+              <span class="{{getItemCssClassByStatus(row[column.key])}}"> {{row[column.key] ? row[column.key]: '-- --'}}
+              </span>
+            </mat-cell>
+          </span>
+          <span *ngIf="column.type == 'action'">
+            <mat-cell *matCellDef="let user">
+              <button (click)="viewDetails(user)" matTooltip="View Details" class="actions"
+                mat-button>View Details</button> &nbsp;
+            </mat-cell>
+          </span>
+        </ng-container>
+      </span>
+      <mat-header-row *matHeaderRowDef="displayedColumns;sticky: true" class="p-button">
+      </mat-header-row>
+      <mat-row *matRowDef="let row; columns: displayedColumns;">
+      </mat-row>
+    </mat-table>
+
+    <div class="message" *ngIf="noData || !dataSource">No records found</div>
+    <mat-paginator [length]="recordCount" [showFirstLastButtons]="true" [pageSize]="10"
+      [pageSizeOptions]="[5, 10, 25, 100]">
```

Comment:
Add this as a constant in the core module and reuse

Reply:
Can you please explain this bro